### PR TITLE
[toplevel] Remove long ago deprecated and NOOP options.

### PR DESCRIPTION
**Pull request by ejgallego**
Minor clean up, no sense in having these as they do nothing.

### CODE REVIEW

**Comment by maximedenes** [comment]
The `-byte` snd `-opt` options for `coqc` might still be useful, and they do something. So this PR may have to be refined.

**Comment by ejgallego** [comment]
Good catch, updated.

File: stm/asyncTaskQueue.ml [hunk 1]
```diff
@@ -123,7 +123,7 @@ module Make(T : Task) = struct
                  "-worker-id"; name;
                  "-async-proofs-worker-priority";
                    Flags.string_of_priority !Flags.async_proofs_worker_priority]
-        | ("-ideslave"|"-emacs"|"-emacs-U"|"-batch")::tl -> set_slave_opt tl
```
Comment (maximedenes) [inline stm/asyncTaskQueue.ml]: Was `-emacs-U` really a noop here? Isn't used by PG btw?

Comment (ejgallego) [inline stm/asyncTaskQueue.ml]: That is a good point, @Matafou @cpitclaudel , why does PG need two options `-emacs` vs `-emacs-U` ?

Comment (psteckler) [inline stm/asyncTaskQueue.ml]: Vanilla PG uses `-emacs` these days.

Comment (ejgallego) [inline stm/asyncTaskQueue.ml]: The option is already marked as obsolete below; however PG may need a minor (one line) patch.

Comment (maximedenes) [inline stm/asyncTaskQueue.ml]: Ok but this it is the second time that I find a non-noop option removed by this PR, despite what the title claims. So I think it should be reviewed (I had only a very brief look).

Comment (ejgallego) [inline stm/asyncTaskQueue.ml]: > Vanilla PG uses -emacs these days.

It still has an occurrence but it is a bug.

**Comment by ejgallego** [comment]
> Ok but this it is the second time that I find a non-noop option removed by this PR, despite what the title claims. So I think it should be reviewed (I had only a very brief look).

This is not accurate as this time the removal of `-emacs-U` is correct, it is PG who has the bug here.

**Comment by maximedenes** [comment]
> This is not accurate as this time the removal of -emacs-U is correct, it is PG who has the bug here.

Do you mean that `-emacs-U` was a noop? I haven't checked.

**Comment by ejgallego** [comment]
It was not a noop but indeed printed a warning message. No one was using it since ages (except for the leftover in PG) What I mean is this option should indeed go, contrary to the first case where we had an error.

**Comment by maximedenes** [comment]
Ok, but I don't see what is not inaccurate in my claim above that despite the title of this PR, it removed some non noop options. So please make either the title accurate, and get an ok for the removal of non noop options.

**Comment by maximedenes** [comment]
It's not a big deal to make small errors like that, we all do, but denying them doesn't help reviewing... In this case, a list of non noop options removed by this PR would allow experts to review it in a minute.

**Comment by psteckler** [comment]
What did coqtop do with `emacs-U`, versus `-emacs`?

**Comment by ejgallego** [comment]
Well, indeed the title is wrong in the case of this option.

**Comment by ejgallego** [comment]
It should be `quasi` NOOPs, I dunno, this cleanup stuff is very tricky.

**Comment by maximedenes** [comment]
If we want to get done with this quickly, maybe someone (@Zimmi48 ?) can have another quick look. If some "quasi-noop" options are found, we just make a list and check they are not used to the best of our knowledge. Then we merge.

**Comment by Zimmi48** [comment]
Well, from what I see the options removed in this PR were all deprecated, so I would just change the PR title and commit message to "Remove deprecated options of coqtop and coqmktop."

Most of them were clearly no-ops, with the exception of `-emacs-U` and `-notactics`. I don't know what `-notactics` was supposed to do, but is it possible that there is more dead code lurking after

```diff
-let remove_top_ml () = Mltop.remove () 
```

?

**Comment by silene** [comment]
> but is it possible that there is more dead code lurking?

No, running Coq without a toplevel is the default behavior these days, so the underlying code is necessarily used. What would be interesting to know, however, is whether there are still people running Coq with a toplevel. If not, there would be quite a bit of dead code to remove.

**Comment by psteckler** [comment]
I changed vanilla PG to use `-emacs` where it had been using `-emacs-U`. (Also, I removed code that filtered out that obsolete flag.)

File: stm/asyncTaskQueue.ml (hunk 1 above)
Comment (herbelin) [inline stm/asyncTaskQueue.ml]: AFAIR `-emacs` was using characters >= 128 for the emacs communication (e.g. 249) and this was broking transferring utf8. The option `-emacs-U` was introduced to support utf8 (f8d64ae9). After a transition period, `-emacs` started itself to call the code of `-emacs-U` and `-emacs-U` be deprecated (77f1db2).

**Comment by Matafou** [comment]
-emacs-U is n.indeed useless. FTR it was about Unicode in emacs vs special
chars used in pg. this was fixed ages ago.
P

Le mer. 19 juil. 2017 à 17:45, psteckler <notifications@github.com> a
écrit :

**Comment by ejgallego** [comment]
Commit and PR renamed.

**Comment by ejgallego** [comment]
Signalling this one for consideration, just in case.

**Pull request merged**